static-assets/components/cstudio-dialogs/uploadCMIS-dialog.js

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright (C) 2007-2019 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+var YDom = YAHOO.util.Dom;
+var YEvent = YAHOO.util.Event;
+
+CStudioAuthoring.Dialogs = CStudioAuthoring.Dialogs || {};
+
+/**
+ * Submit to go live
+ */
+CStudioAuthoring.Dialogs.UploadCMISDialog = CStudioAuthoring.Dialogs.UploadCMISDialog || {
+
+    /**
+     * initialize module
+     */
+    initialize: function(config) {
+    },
+
+    /**
+     * show dialog
+     */
+    showDialog: function(site, path, repositoryId, serviceUri, callback) {
+        this._self = this;
+        console.log('dialog');
+
+        this.dialog = this.createDialog(path, site, repositoryId, serviceUri);
+
+        this.site = site;
+        this.path = path;
+        this.profile = repositoryId;
+        this.asPopup = true;
+        this.serviceUri = serviceUri;
+        this.callback = callback;
+        this.dialog.show();
+        document.getElementById("cstudio-wcm-popup-div_h").style.display = "none";
+
+        if(window.frameElement){
```

Comment:
Is this dialog always going to be on the same domain? 
Accessing parent/child windows is usually calling for trouble.

Reply:
FrameElement is getting the iframe which the element is. I don't think the domain is going to change because it is a dialog that it is open inside from engine.